src/vma/sock/sockinfo_udp.cpp

```diff
@@ -563,6 +563,14 @@ int sockinfo_udp::connect(const struct sockaddr *__to, socklen_t __tolen)
 		}
 		BULLSEYE_EXCLUDE_BLOCK_END
 		on_sockname_change(name, *namelen);
+
+		if (m_rx_ring_map.size() == 1) {
+			rx_ring_map_t::iterator rx_ring_iter = m_rx_ring_map.begin();
+			m_p_rx_ring = rx_ring_iter->first;
+		} else {
+			si_udp_logdbg("ring map size: %d", m_rx_ring_map.size());
+		}
+
```

Comment:
According current flow for TCP we do this in pair with attach_as_uc_receiver() call as
```
    attach_as_uc_receiver();
	if (m_rx_ring_map.size() == 1) {
		rx_ring_map_t::iterator rx_ring_iter = m_rx_ring_map.begin();
		m_p_rx_ring = rx_ring_iter->first;
	}
```

I am suggesting to do the same for UDP. Please consider moving these code to sockinfo_udp::on_sockname_change() from this place and from bind.